File: Fragments/app/src/main/java/org/guildsa/fragments/TextEditFragment.java
package org.guildsa.fragments;

import android.app.Activity;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;
import android.widget.SeekBar;

public class TextEditFragment extends Fragment implements SeekBar.OnSeekBarChangeListener {

    private static int seekProgess = 20;
    private static EditText editText;

    TextEditListener activityCallback;

    public interface TextEditListener {
        public void onTextEditButtonClick(int fontsize, String text);
    }

    @Override
    public void onAttach(Activity activity) {

        super.onAttach(activity);

        try {
            activityCallback = (TextEditListener) activity;
        } catch (ClassCastException e) {
            throw new ClassCastException(activity.toString() + " must implement ToolbarListener");
        }
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        // Inflate the layout for this fragment
        View view =  inflater.inflate(R.layout.text_edit_fragment, container, false);

        editText = (EditText) view.findViewById(R.id.editText1);
        editText.setText("New Text");

        final SeekBar seekbar = (SeekBar) view.findViewById(R.id.seekBar1);
        seekbar.setProgress(seekProgess);
        seekbar.setOnSeekBarChangeListener(this);

        final Button button = (Button) view.findViewById(R.id.button1);
        button.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                buttonClicked(v);
            }
        });

        return view;
    }

    public void buttonClicked(View view) {

        activityCallback.onTextEditButtonClick(seekProgess, editText.getText().toString());
    }

    @Override
    public void onProgressChanged(SeekBar seekBar, int progress, boolean fromUser) {
        seekProgess = progress;
    }

    @Override
    public void onStartTrackingTouch(SeekBar seekBar) {
    }

    @Override
    public void onStopTrackingTouch(SeekBar seekBar) {
    }
}

File: Fragments/app/src/main/java/org/guildsa/fragments/TextEditFragment.kt
package org.guildsa.fragments

import android.app.Activity
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.EditText
import android.widget.SeekBar
import android.widget.SeekBar.OnSeekBarChangeListener
import androidx.fragment.app.Fragment

class TextEditFragment : Fragment(), OnSeekBarChangeListener {
    private var seekProgess = 20
    private var editText: EditText? = null
    private var parentActivity: TextEditListener? = null

    interface TextEditListener {
        open fun onTextEditButtonClick(fontSize: Int, text: String?)
    }

    override fun onAttach(activity: Activity?) {
        super.onAttach(activity)
        // If our fragment is attached to an Activity, cache a reference to the Activity that is
// using our TextEditFragment and make sure that it implements the TextEditListener
// interface. If it doesn't implement TextEditListener, we have a problem!
        parentActivity = try {
            activity as TextEditListener?
        } catch(e: ClassCastException) {
            throw ClassCastException(activity.toString() + " must implement TextEditListener")
        }
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? { // Inflate the layout for this fragment
        val view = inflater.inflate(R.layout.text_edit_fragment, container, false)
        editText = view.findViewById<View?>(R.id.editText1) as EditText?
        editText.setText("New Text")
        val seekbar = view.findViewById<View?>(R.id.seekBar1) as SeekBar?
        seekbar.setProgress(seekProgess)
        seekbar.setOnSeekBarChangeListener(this)
        val button = view.findViewById<View?>(R.id.button1) as Button?
        button.setOnClickListener(View.OnClickListener { v -> buttonClicked(v) })
        return view
    }

    fun buttonClicked(view: View?) { // When the button on our fragment is clicked, tell the Activity that is using us
// that the button was clicked and pass the required info.
        parentActivity.onTextEditButtonClick(seekProgess, editText.getText().toString())
    }

    override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
        seekProgess = progress
    }

    override fun onStartTrackingTouch(seekBar: SeekBar?) {}
    override fun onStopTrackingTouch(seekBar: SeekBar?) {}
}